### PR TITLE
RDM-11523 unauthorised users could update some collections

### PR DESCRIPTION
**Pull request by RebeccaBaker**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-11523

### Change description ###
 unauthorised users could update details for collections of CCD base types

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by MSancaktutar — APPROVED** [review]
LGTM

File: src/test/java/uk/gov/hmcts/ccd/domain/service/common/CompoundAccessControlServiceTest.java [hunk 1]
```diff
@@ -1113,6 +1118,181 @@ void shouldDenyAccessWhenChildUpdatedAndNoFineGrainedACLExists() throws IOExcept
...
+                + "                \"SubNestedField2\": \"SomethingToBeIgnored\"\n"
+                + "            }\n"
+                + "        }\n"
+                + "    ]\n"
+                + "  }\n"
+                + "}");
+
+            assertThat(compoundAccessControlService.hasAccessForAction(
+                newDataNode, existingDataNode, caseTypeDefinition.getCaseFieldDefinitions().get(0),
+                USER_ROLES), is(false));
```
Comment (danlysiak) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/common/CompoundAccessControlServiceTest.java:1293]: Think something is wrong with the set up of this test, as ACL is actually all true but access is still being denied?

**Review by danlysiak — APPROVED** [review]
Implementation looks good, minor comment on test.

**Comment by MSancaktutar** [comment]
Merged as confirmed.

**Pull request merged**